GoogleForCreators/web-stories-wp
# Ad Manager & AdSense support

## Summary

Add basic Ad Manager & AdSense integration. 

## Relevant Technical Choices

Add basic PHP classes to add amp ad tags. 

There are currently no tests for the dropdown, as this is pretty hard to test and no existing use of the dropdown has tests. 

There are no e2e tests, as the javascript and php unit tests should cover most error states. 

## To-do



## User-facing changes

<img width="967" alt="Screenshot 2020-12-18 at 11 00 51" src="https://user-images.githubusercontent.com/841956/102627569-aa90d000-4148-11eb-9872-a733a7ae6135.png">
<img width="993" alt="Screenshot 2020-12-18 at 11 01 03" src="https://user-images.githubusercontent.com/841956/102627572-ab296680-4148-11eb-951e-d7de38271578.png">
<img width="482" alt="Screenshot 2020-12-18 at 11 00 40" src="https://user-images.githubusercontent.com/841956/102627574-abc1fd00-4148-11eb-850e-821f8c9ebfcb.png">


## Testing Instructions

1. On the Settings screen, see the new "Monetization" section
2. Select "AdSense" and enter ID values in the same format as in the examples  
  (Other formats should cause warnings / red borders)
3. Save changes and view a story on the frontend. Verify that there's an `<amp-story-auto-ads>` tag in the page source for _AdSense_
4. On the settings screen, select "Ad Manager" now, verifying that the AdSense options will be hidden
5. Enter a slot ID in the same format as in the example  
  (Other formats should cause warnings / red borders)
6. Save changes and view a story on the frontend. Verify that there's an `<amp-story-auto-ads>` tag in the page source for _Ad Manager_, but none for _AdSense_

---



Fixes #5700
Fixes #4803
